tests/test_graph.cpp

```diff
@@ -0,0 +1,194 @@
+#include <array>
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+#include <cstring>
+#include <iostream>
+#include <string>
+
+#include <cudawrappers/cu.hpp>
+#include <cudawrappers/nvrtc.hpp>
+
+TEST_CASE("Test cu::Graph", "[graph]") {
+  const std::string kernel = R"(
+    extern "C" __global__ void vector_print(float *a, size_t array_size) {
+      int i = blockIdx.x * blockDim.x + threadIdx.x;
+      if (i < array_size) {
+#ifdef DEBUG
+        printf("a[%d] = %f\n", i, a[i]);
+#endif
+        a[i] *= 2.0f;
+      }
+    }
+  )";
+
+  cu::init();
+  cu::Device device(0);
+  cu::Context context(CU_CTX_SCHED_BLOCKING_SYNC, device);
+  context.setCurrent();
+  nvrtc::Program program(kernel, "kernel.cu");
+  program.compile({});
+  cu::Module module(static_cast<const void*>(program.getPTX().data()));
+
+  cu::Stream stream;
+  const size_t array_size = 1000;
+  SECTION("Test cu::hostNode single value") {
+    auto fn = [](void* data) {
+      int* ptr = static_cast<int*>(data);
+      std::cout << "Host node content: " << *ptr << std::endl;
```

Comment:
Put inside `DEBUG` guards, or consider removing the print statements altogether (also the other ones below).
If you decide to do so, also remove the include of `iostream`.
```suggestion
      std::cout << "Host node content: " << *ptr << std::endl;
```